stingray/gti.py

```diff
@@ -1168,10 +1168,13 @@ def gti_border_bins(gtis, time, dt=None, epsilon=0.001):
         dt = np.median(np.diff(time))
 
     epsilon_times_dt = epsilon * dt
-    spectrum_start_bins = np.array([], dtype=int)
-    spectrum_stop_bins = np.array([], dtype=int)
+
+    spectrum_start_bins = []
+    spectrum_stop_bins = []
+
     for g in gtis:
-        good = (time - dt / 2 >= g[0]) & (time + dt / 2 <= g[1])
+        good = ((time - dt / 2 >= g[0] - epsilon_times_dt) &
+                (time + dt / 2 <= g[1] + epsilon_times_dt))
```

Comment:
What does this change do, especially in the context of the simulation of arrival times? I can't really figure out what's happening here.

Reply:
It's a small numerical correction, since sometimes timestamps can be a floating point epsilon outside or inside the GTI, and that makes light curve intervals of different length in different segments. Those epsilons I add are tolerances (the bin edge can be 1/1000 dt outside the GTI, and it will still be included). I had this bug during the work on this PR and I fixed it.